### PR TITLE
feat(forwarder): ERC1155Receiver implementation

### PR DESCRIPTION
**Pull request by ryan-bitgo**
ticket: BG-39187

### CODE REVIEW

File: contracts/Forwarder.sol [hunk 1]
```diff
@@ -95,6 +97,44 @@ contract Forwarder is ReentrancyGuard, ERC721TokenReceiver {
...
+    address _operator,
+    address _from,
+    uint256 id,
+    uint256 value,
+    bytes calldata data
+  ) external virtual override nonReentrant returns (bytes4) {
+    IERC1155 instance = IERC1155(msg.sender);
+
+    if (autoFlush721) {
+      instance.safeTransferFrom(address(this), parentAddress, id, value, data);
```
Comment (twtaylorbitgo) [inline contracts/Forwarder.sol:110]: any checks we could add here? possibly that msg.sender implements ERC165 for ERC1155?

File: contracts/Forwarder.sol [hunk 2]
```diff
@@ -95,6 +97,44 @@ contract Forwarder is ReentrancyGuard, ERC721TokenReceiver {
...
+  }
+
+  function onERC1155BatchReceived(
+    address _operator,
+    address _from,
+    uint256[] calldata ids,
+    uint256[] calldata values,
+    bytes calldata data
+  ) external virtual override nonReentrant returns (bytes4) {
+    IERC1155 instance = IERC1155(msg.sender);
```
Comment (twtaylorbitgo) [inline contracts/Forwarder.sol:123]: same bitmask check above

**Review by twtaylorbitgo — CHANGES_REQUESTED** [review]
we'll need to add an explicit flush function as well for both batch and single, just in case you turn off flushing and want to flush manually

File: contracts/Forwarder.sol (hunk 1 above)
Comment (ryan-bitgo) [reply contracts/Forwarder.sol:114]: fixed

File: contracts/Forwarder.sol [hunk 3]
```diff
@@ -95,6 +97,52 @@ contract Forwarder is ReentrancyGuard, ERC721TokenReceiver {
...
+    );
+
+    if (autoFlush721) {
+      instance.safeBatchTransferFrom(
+        address(this),
+        parentAddress,
+        ids,
+        values,
+        data
+      );
```
Comment (twtaylorbitgo) [inline contracts/Forwarder.sol:140]: can we `require` this and if it doesn't succeed, rollback?

**Review by twtaylorbitgo — CHANGES_REQUESTED** [review]

File: contracts/Forwarder.sol [hunk 4]
```diff
@@ -1,6 +1,8 @@
 // SPDX-License-Identifier: Apache-2.0
 pragma solidity 0.7.5;
 import '@uniswap/lib/contracts/libraries/TransferHelper.sol';
+import '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
```
Comment (barathbitgo) [inline contracts/Forwarder.sol:4]: unused

Comment (ryan-bitgo) [reply contracts/Forwarder.sol:4]: used here: `instance.supportsInterface(type(IERC1155).interfaceId)`

File: contracts/Forwarder.sol (hunk 3 above)
Comment (ryan-bitgo) [reply contracts/Forwarder.sol:145]: function should revert if it fails. doesnt return true or false

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/IERC1155.sol#L99

**Review by twtaylorbitgo — APPROVED** [review]

**Pull request merged**